sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1294,15 +1327,12 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
         return connection;
       }
     }
-    lock.unlock();
```

Comment:
Are we removing this because `lock` will automatically get unlocked once out of scope? Might be worth adding a comment at end of scope to make it easier to track.

Reply:
That is correct - the call to `lock.unlock()` provides absolutely no value.

THe code already says:
```cpp
   // Critical section. Needs to own ConnectionPoolMutex before executing
    // Lock mutex to access connection pool. mutex is unlock as soon as lock is out of scope
 ```

IMHO this is sufficient to explain what's happening. If it's not, could you please suggest your proposed text which explains this better?